rpcs3/Emu/Cell/lv2/sys_usbd.cpp

```diff
@@ -525,24 +574,37 @@ usb_handler_thread::~usb_handler_thread()
 	handled_devices.clear();
 	open_pipes.clear();
 	usb_devices.clear();
+	usb_passthrough_devices.clear();
 
 	for (u32 index = 0; index < MAX_SYS_USBD_TRANSFERS; index++)
 	{
 		if (transfers[index].transfer)
 			libusb_free_transfer(transfers[index].transfer);
 	}
 
+#if LIBUSB_API_VERSION >= 0x01000102
+	libusb_hotplug_deregister_callback(ctx, callback_handle);
+#endif
+
 	if (ctx)
 		libusb_exit(ctx);
 }
 
 void usb_handler_thread::operator()()
 {
 	timeval lusb_tv{0, 0};
-
+	u64 usb_hotplug_timeout = 0;
 	while (ctx && thread_ctrl::state() != thread_state::aborting)
 	{
-		// Todo: Hotplug here?
+		const u64 now = get_system_time();
+		if ((now > usb_hotplug_timeout && !hotplug_supported) || (received_hotplug_event_timeout && now > received_hotplug_event_timeout))
```

Comment:
I feel there is a simplification possible here, we don't really need 2 values to keep track of when to scan, one value should be enough.
If hotplug is not supported, value should be initialized at the start of the thread to now + 4s. If hotplug is supported it should be left 0 and set by the callback. Then you can do `if (hotplug_event_timeout && now > hotplug_event_timeout)` then set it to 0 if hotplug is supported set it to 0 otherwise set it to now + 4s.